### PR TITLE
Fix failing filesystem monitor, and upgrade plots

### LINKED ISSUE #1087: Monitor_filesystem crashes due to file with no suffix
From the log of a trial run today:

```
monitor_filesystem_pljwql1.stsci.edu_2022-11-07-14-21.log

    func(*args, **kwargs)
  File "logging_functions.py", line 255, in wrapped
    func(*args, **kwargs)
  File "monitor_filesystem.py", line 280, in monitor_filesystem
    general_results_dict, instrument_results_dict = gather_statistics(general_results_dict, instrument_results_dict)
  File "monitor_filesystem.py", line 105, in gather_statistics
    filetype = filename_dict['suffix']
KeyError: 'suffix'
```
There's no mention of which file specifically caused the problem. 

We can probably get around this with a try/except and skip any files with no suffixes. Looking into the filesystem, it looks like gather_statistics looks at all files (e.g. MAST-hosted thumbnails, association files, ecsv files from level 3). Does it make sense to track all of these and add them to the list of new files in the filesystem? I guess it does if you want to know the total size of the filesystem. But if the goal is to see how many new rootnames are present, then you're counting too much. Perhaps that is a larger question for later though.

### PR DESCRIPTION
**Pull request by bhilbert4**
Resolves #1087 

At the moment, monitor_filesystem.py is crashing when trying to parse a file in the filesystem that has no suffix. This PR adds a try/except around the parser results such that if a file has no suffix, it is skipped in the filesystem counting.

### CODE REVIEW

**Comment by bhilbert4** [comment]
- [x] Still needs to be tested on the dev or test server

**Comment by bhilbert4** [comment]
Test run on the dev server: so far it looks like the files causing the problems are msa files (e.g. jw02640001001_02_msa.fits)

For these files, the filename parser returns with:
```
{'program_id': '02640',
 'observation': '001',
 'visit': '001',
 'filename_type': 'stage_2_msa',
 'instrument': 'nirspec'}
```

**Comment by bhilbert4** [comment]
Another crash:
11/07/2022 17:07:23 PM CRITICAL: Traceback (most recent call last):
  File "logging_functions.py", line 290, in wrapped
    func(*args, **kwargs)
  File "logging_functions.py", line 255, in wrapped
    func(*args, **kwargs)
  File "monitor_filesystem.py", line 286, in monitor_filesystem
    general_results_dict, instrument_results_dict = gather_statistics(general_results_dict, instrument_results_dict)
  File "monitor_filesystem.py", line 96, in gather_statistics
    general_results_dict['total_file_size'] += os.path.getsize(file_path)
  File "lib/python3.8/genericpath.py", line 50, in getsize
    return os.stat(filename).st_size
FileNotFoundError: [Errno 2] No such file or directory: 'public/jw02561/jw02561001002/dads1095711200654586864.tmp'

**Comment by bhilbert4** [comment]
@mfixstsci this is ready for review.

**Comment by bhilbert4** [comment]
Plots of files per filter, files per filetype, and total file count per instrument have been updated. The plots of disk storage have also been tweaked to make them more readable.

File: jwql/utils/utils.py [hunk 1]
```diff
@@ -653,3 +653,31 @@ def grouper(iterable, chunksize):
         if not chunk:
             return
         yield chunk
+
+def map_filterpupil_name_to_wavelength(filterpupil):
+    """Note that this is pretty nircam-specific at the moment. need to generalize a bit for other instruments
+    """
+    wlp4_wave = 2.12
+    dark_fake_wave = 0.6
+    for e in filterpupil:
```
Comment (mfixstsci) [inline jwql/utils/utils.py]: What does `e` represent here? Maybe we add the parameter portion of the docstring here to clear up any confusion.

File: jwql/jwql_monitors/monitor_filesystem.py [hunk 2]
```diff
@@ -66,6 +69,35 @@
 OUTPUTS = SETTINGS['outputs']
 
 
+def files_per_filter():
```
Comment (mfixstsci) [inline jwql/jwql_monitors/monitor_filesystem.py:72]: Thanks @bhilbert4 when I first started the dashboard, I was hoping there was a table that had this information available and easy to find. This is great.

File: jwql/jwql_monitors/monitor_filesystem.py [hunk 3]
```diff
@@ -93,31 +125,45 @@ def gather_statistics(general_results_dict, instrument_results_dict):
...
+                        if filetype is not None:
+                            # Populate instrument specific stats
+                            if instrument not in instrument_results_dict:
+                                instrument_results_dict[instrument] = {}
+                            if filetype not in instrument_results_dict[instrument]:
+                                instrument_results_dict[instrument][filetype] = {}
+                                instrument_results_dict[instrument][filetype]['count'] = 0
+                                instrument_results_dict[instrument][filetype]['size'] = 0
+                            instrument_results_dict[instrument][filetype]['count'] += 1
+                            instrument_results_dict[instrument][filetype]['size'] += os.path.getsize(file_path) / (2**40)
```
Comment (mfixstsci) [inline jwql/jwql_monitors/monitor_filesystem.py:166]: Does `/2**40` convert bytes to TB or something like that?

File: jwql/website/apps/jwql/bokeh_dashboard.py [hunk 4]
```diff
@@ -277,7 +631,7 @@ def dashboard_anomaly_per_instrument(self):
         """
 
         # Set title and figures list to make panels
-        title = 'Anamoly Types per Instrument'
+        title = 'Anomaly Types per Instrument'
```
Comment (mfixstsci) [inline jwql/website/apps/jwql/bokeh_dashboard.py:585]: Man, I really nailed the spelling on this one....

File: jwql/website/apps/jwql/bokeh_dashboard.py [hunk 5]
```diff
@@ -259,14 +347,280 @@ def dashboard_exposure_count_by_filter(self):
...
+            # commented out for testing
+            #instrument = data.loc[i]['instrument']
+            #filterpupil = data.loc[i]['filter_pupil']
+            #num_obs = data.loc[i]['obs_per_filter_pupil']
+
+            # For testing
+            instrument = 'nircam'
+            #num_obs = np.linspace(0.01, 30., 100)
+            #num_obs = np.linspace(1, 400, 100)
+            num_obs = np.array([88, 3, 95, 187, 1, 1, 1, 60, 10, 10, 6, 175, 3, 56, 32, 35, 8, 5, 1, 36, 8, 4, 12, 43, 3, 3, 8, 15, 255, 1, 12, 6, 12, 6, 9, 115, 19, 17, 14, 560, 6, 3, 5, 3, 241, 273, 61, 8, 6, 10, 9, 219, 1, 12, 6, 5, 2, 45, 8, 6, 5, 9, 57, 50, 1, 29, 66, 6, 109, 6, 8, 6, 35, 75, 216, 16, 36, 4, 39, 37, 8, 6, 5, 6, 73, 10, 7, 8, 9, 38, 10, 9, 8, 13, 202, 315, 27, 44, 14, 38, 5, 39, 29, 10, 9, 8, 6, 327, 10, 10, 9, 6, 253,109,5,57,57])
```
Comment (mfixstsci) [inline jwql/website/apps/jwql/bokeh_dashboard.py]: Why do we have all of the number of observations hard coded here?

**Review by mfixstsci — APPROVED** [review]
Hey @bhilbert4 thank you for the updates to the dashboard. I know this has been lingering around since data started rolling in.

File: jwql/jwql_monitors/monitor_filesystem.py (hunk 3 above)
Comment (bhilbert4) [reply jwql/jwql_monitors/monitor_filesystem.py:166]: Yeah, that's right. I copied that over from the old funtciton.

File: jwql/utils/utils.py (hunk 1 above)
Comment (bhilbert4) [reply jwql/utils/utils.py]: Oh shoot. I forgot about this function. I'll have to look and make sure it's not forgetting anything from the other instruments as well.

File: jwql/website/apps/jwql/bokeh_dashboard.py (hunk 4 above)
Comment (bhilbert4) [reply jwql/website/apps/jwql/bokeh_dashboard.py:585]: 😆

File: jwql/website/apps/jwql/bokeh_dashboard.py (hunk 5 above)
Comment (bhilbert4) [reply jwql/website/apps/jwql/bokeh_dashboard.py]: Oops. This is leftover from testing. The "commented out for testing" lines above need to be used instead.

Comment (mfixstsci) [reply jwql/website/apps/jwql/bokeh_dashboard.py]: Thank you!

File: jwql/website/apps/jwql/bokeh_dashboard.py [hunk 6]
```diff
@@ -259,14 +349,276 @@ def dashboard_exposure_count_by_filter(self):
...
+        #start_angle = np.arange(len(waves))*360./len(waves)
+        #end_angle = np.arange(1, len(waves)+1)*360./len(waves)
+
+        # This is a loop over instruments
+        for i in range(len(data)):
+            instrument = data.loc[i]['instrument']
+            filterpupil = data.loc[i]['filter_pupil']
+            num_obs = data.loc[i]['obs_per_filter_pupil']
+
+            # For testing
```
Comment (mfixstsci) [inline jwql/website/apps/jwql/bokeh_dashboard.py]: Thanks for commenting this out @bhilbert4 are we keeping this in the code base for future testing? I know we discussed using a tree plot here so I am guessing that you will want this when we do decided to update the bokeh versions?

**Comment by bhilbert4** [comment]
Monitor is currently failing. It looks like there are a large number of bad symlinks in the filesystem, all for level3 products. Since JWQL doesn't currently do anything with level 3 data, it seems like we can skip over these. 

Also, now that I think about it, I'm not sure why we care about the total size of the filesystem, since we are looking directly at MAST, as opposed to our own copy of the data.

**Comment by bhilbert4** [comment]
@mfixstsci 2 tests are still in progress, but the others have all passed, so I'm hopeful this is all set now.

**Review by mfixstsci — APPROVED** [review]
Awesome, thank you for the changes.

File: jwql/website/apps/jwql/bokeh_dashboard.py (hunk 6 above)
Comment (bhilbert4) [reply jwql/website/apps/jwql/bokeh_dashboard.py]: Oops. I forgot to remove the test data. I can strip that out. 

That's correct about the treeplot code though. I'm banking on being able to turn this on as soon as we are able to switch to bokeh > 3.0. At that point the tree plot will replace the pie chart.

Comment (bhilbert4) [reply jwql/website/apps/jwql/bokeh_dashboard.py]: Ok, I've stripped away the commented out lines.

**Review by mfixstsci — APPROVED** [review]
Thank you!

**Pull request merged**